### PR TITLE
Fix torch.distributed._sharded_tensor DeprecationWarning

### LINKED ISSUE #13256: DeprecationWarning: torch.distributed._sharded_tensor
## 🐛 Bug

The latest versions of pytorch-lightning and PyTorch give the following deprecation warning:
```
/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pytorch_lightning/core/lightning.py:118: in __init__
    self._register_sharded_tensor_state_dict_hooks_if_available()
/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pytorch_lightning/core/lightning.py:2067: in _register_sharded_tensor_state_dict_hooks_if_available
    from torch.distributed._sharded_tensor import pre_load_state_dict_hook, state_dict_hook
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    # Keep old package for BC purposes, this file should be removed once
    # everything moves to the `torch.distributed._shard` package.
    import sys
    import torch
    import warnings
    from torch.distributed._shard.sharded_tensor import *  # noqa: F403
>   warnings.warn(
        "torch.distributed._sharded_tensor will be deprecated, use torch.distributed._shard.sharded_tensor instead",
        DeprecationWarning
    )
E   DeprecationWarning: torch.distributed._sharded_tensor will be deprecated, use torch.distributed._shard.sharded_tensor instead
```
This chunk of code has since moved, but `torch.distributed._sharded_tensor` is still being imported in `pytorch_lightning/core/module.py` and `tests/core/test_lightning_module.py`.

### To Reproduce

Run code using `LightningModule` through pytest with the latest version of PyTorch.

### Expected behavior

No DeprecationWarning.

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): 1.6.4
- PyTorch Version (e.g., 1.10): 1.11.0
- Python version (e.g., 3.9): 3.10.4, etc.
- OS (e.g., Linux): Linux, etc.
- CUDA/cuDNN version: N/A
- GPU models and configuration: N/A
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`: N/A
- Any other relevant information: N/A

### PR DESCRIPTION
**Pull request by adamjstewart**
## What does this PR do?

Fixes #13256. The refactor was done in https://github.com/pytorch/pytorch/pull/72141 which was first introduced in PyTorch 1.11.0. @akihironitta

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Comment by adamjstewart** [comment]
Resolved the merge conflicts, should be ready to merge again.

**Review by awaelchli — APPROVED** [review]
